src/Context/Auth/AuthProvider.jsx

```diff
@@ -0,0 +1,56 @@
+import { createContext, useContext, useReducer } from "react";
+
+const initialAuthState = {
+  login: false,
+  token: "",
+  user: {
+    firstName: "",
+    lastName: "",
+    email: "",
+    dp: "",
+  },
+};
+
+const authReducer = (auth, action) => {
+  switch (action.type) {
+    case "login":
+      return {
+        ...auth,
+        login: true,
+        token: action.payload.encodedToken,
+        user: {
+          firstName: action.payload.foundUser.firstName,
+          lastName: action.payload.foundUser.lastName,
+          email: action.payload.foundUser.email,
+          dp:
+            action.payload.foundUser.firstName.slice(0, 1) +
```

Comment:
Instead of slice just use [0]

Reply:
Thank you for pointing it out. Will update it.